containers/src/Kokkos_DynRankView.hpp

```diff
@@ -146,14 +147,19 @@ struct DynRankDimTraits {
     } else
 #endif
       new_layout.stride = layout.stride;
+    if constexpr (std::is_same_v<Layout, Kokkos::LayoutRight>) {
+      if (new_rank != unspecified && new_rank > 0 &&
+          layout.dimension[new_rank - 1] == layout.stride)
+        new_layout.stride = unspecified;
+    }
     return new_layout;
   }
 
   // LayoutStride
   template <typename Layout>
   KOKKOS_INLINE_FUNCTION static std::enable_if_t<
       (std::is_same_v<Layout, Kokkos::LayoutStride>), Layout>
-  createLayout(const Layout& layout) {
+  createLayout(const Layout& layout, [[maybe_unused]] size_t new_rank = -1) {
```

Comment:
```suggestion
  createLayout(const Layout& layout, [[maybe_unused]] size_t new_rank = unspecified) {
```

should this match the default arg for `new_rank` like L123?

Reply:
Yes